aws-s3-transfer-manager/src/operation/download.rs

```diff
@@ -70,7 +73,8 @@ impl Download {
             todo!("single part download not implemented")
         }
 
-        let ctx = DownloadContext::new(handle);
+        let bucket_type = BucketType::from_bucket_name(input.bucket().unwrap_or(""));
```

Comment:
Is the bucket name required for `DownloadInput` so we can `.expect` on `input.bucket()`, rather than create a `BucketType` from an empty string?

Reply:
It looks like we validate on two operations but not the others:

* https://github.com/awslabs/aws-s3-transfer-manager-rs/blob/main/aws-s3-transfer-manager/src/operation/download_objects/input.rs#L210
* https://github.com/awslabs/aws-s3-transfer-manager-rs/blob/main/aws-s3-transfer-manager/src/operation/upload_objects/input.rs#L102

Probably an oversight